compiler/test/fail_compilation/fail158.d

```diff
@@ -1,7 +1,7 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/fail158.d(17): Error: more initializers than fields (2) of `S`
+fail_compilation/fail158.d(17): Error: too many initializers for `S`
```

Comment:
The new error is less informative.

Reply:
I added the field count in the `too many initializers` error message.